File: frontend/src/types/connection.ts

export type Connection = {
  connectionId: string;
  webId: string;
  description: string;
  fromSourceId: string;
  toSourceId: string;
  created: Date;
  updated: Date;
  aiGenerated?: boolean;
};

export type CreateConnection = {
  webId: string;
  fromSourceId: string;
  toSourceId: string;
  description: string;
};

export type UpdateConnection = {
  name: string;
  description: string;
};

export type DeleteConnection = {
  connectionId: string;
};

File: backend/src/models/connection.py
from pydantic import BaseModel
from typing import Optional
from typing_extensions import TypedDict
from datetime import datetime


class Connection(BaseModel):
    connectionId: str
    webId: str
    description: str
    fromSourceId: str
    toSourceId: str
    created: datetime
    updated: datetime
    aiGenerated: Optional[bool] = False


class CreateConnection(BaseModel):
    description: str
    fromSourceId: str
    toSourceId: str
    webId: str


class UpdateConnection(BaseModel):
    description: Optional[str]
    toSourceId: Optional[str]


class DeleteConnection(BaseModel):
    webId: str
